File: LeetCode/HashMap/MajorityElement.java
import java.util.HashMap;

public class MajorityElement {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> map=new HashMap<>();
        int n=(int)nums.length/2;

        for(int num:nums){
            if(map.containsKey(num)){
                map.put(num,map.get(num)+1);
            }
            else{
                map.put(num,1);
            }
        }

        for(int num:map.keySet()){
            if(map.get(num)>n){
                return num;
            }
        }

        return -1;
    }
}

File: LeetCode/HashMap/MajorityElement.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        map={}
        n=len(nums)//2

        for num in nums:
            if num in map:
                map[num]=map[num]+1
            else:
                map[num]=1

        for num in map:
            if map[num]>n:
                return num
            
        return -1